plugin/storage/es/mappings/mapping.go

```diff
@@ -9,36 +9,70 @@ import (
 	"strings"
 
 	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/config"
 )
 
 // MAPPINGS contains embedded index templates.
 //
 //go:embed *.json
 var MAPPINGS embed.FS
 
-// MappingBuilder holds parameters required to render an elasticsearch index template
+// MappingBuilder holds common parameters required to render an elasticsearch index template
 type MappingBuilder struct {
-	TemplateBuilder              es.TemplateBuilder
-	Shards                       int64
-	Replicas                     int64
-	PrioritySpanTemplate         int64
-	PriorityServiceTemplate      int64
-	PriorityDependenciesTemplate int64
-	PrioritySamplingTemplate     int64
-	EsVersion                    uint
-	IndexPrefix                  string
-	UseILM                       bool
-	ILMPolicyName                string
+	TemplateBuilder es.TemplateBuilder
+	Indices         config.Indices
+	EsVersion       uint
+	UseILM          bool
+	ILMPolicyName   string
+}
+
+// IndexTemplateOptions holds parameters required to render an elasticsearch index template
+type IndexTemplateOptions struct {
+	UseILM        bool
+	ILMPolicyName string
+	IndexPrefix   config.IndexPrefix
+	Shards        int64
+	Replicas      int64
+	Priority      int64
+}
+
+func (mb MappingBuilder) getMappingTemplateOptions(mapping string) *IndexTemplateOptions {
+	mappingOpts := &IndexTemplateOptions{}
+	mappingOpts.UseILM = mb.UseILM
+	mappingOpts.ILMPolicyName = mb.ILMPolicyName
+	mappingOpts.IndexPrefix = mb.Indices.IndexPrefix
```

Comment:
can we NOT make this assignment? This field is being overwritten in L113

Reply:
@yurishkuro L113 doesn't overwrite the field. It just adds a prefix to it if it doesn't exist.

Reply:
what I mean is L113 does an unconditional assignment. You can replace that line with:

```
options.IndexPrefix = mb.Indices.IndexPrefix.Apply("")
```
and then `options.IndexPrefix` can be a plain string because you'd never call Apply on it again, you just pass it to the template (https://github.com/jaegertracing/jaeger/pull/6060/files#r1794418861).

Reply:
ah okay i see! thanks so much for the explanation - this is fixed now